CHEWBBACA/PrepExternalSchema/PrepExternalSchema.py

```diff
@@ -0,0 +1,1111 @@
+
+            # determine alleles that only had hits with low BSR
+            hitting_low = list(set(hitting_low) - set(hitting_high))
+
+            # remove seqids that had high BSR hits because
+            # those alleles are already represented
+            hitting_high = list(set(hitting_high))
+            ids_to_blast = [seqid for seqid in ids_to_blast
+                            if seqid not in hitting_high]
+
+            # determine candidates, alleles that only had hit with BSR
+            # between 0.6 and 0.7
+            rep_candidates = list(set(hotspots) - set(hitting_high))
+            # remove representatives
+            rep_candidates = [seqid for seqid in rep_candidates
+                              if seqid not in representatives]
+
+            # decide which candidate to add as representative
+            representatives = determine_new_representative(rep_candidates,
+                                                           prot_seqs,
+                                                           ids_to_blast,
+                                                           representatives)
+
+            # remove files created for current gene iteration
+            os.remove(rep_file)
+            os.remove(blast_output)
+            os.remove(ids_file)
+
+        # write schema file with all alleles
+        gene_lines = fasta_lines(list(gene_seqs.keys()), gene_seqs)
+        write_list(gene_lines, gene_file)
+
+        # write schema file with representatives
+        representatives = [seqids_map[rep] for rep in representatives]
+        gene_rep_lines = fasta_lines(representatives, gene_seqs)
+        write_list(gene_rep_lines, gene_short_file)
+
+        shutil.rmtree(gene_temp_dir)
+
+    return [invalid_alleles, invalid_genes]
+
+
+def main(external_schema, output_schema, cpu_threads, bsr):
+
+    start = time.time()
+
+    print('\nAdapting schema in the following '
+          'directory:\n{0}'.format(os.path.abspath(external_schema)))
+    print('Number of threads: {0}'.format(cpu_threads))
+    print('BLAST Score Ratio: {0}\n'.format(bsr))
+
+    genes_list = []
+
+    # list FASTA files in input directory
+    # or get list of files from text file
+    if os.path.isdir(external_schema) is True:
+        genes_list = os.listdir(external_schema)
+        # get absolute paths
+        genes_list = [os.path.join(external_schema, file)
+                      for file in genes_list]
+    elif os.path.isfile(external_schema) is True:
+        with open(external_schema, 'r') as infile:
+            genes_list = [file.strip() for file in infile.readlines()]
+
+    # filter files based on suffix
+    file_suffixes = ['.fasta', '.fna', '.ffn']
+    genes_list = filter_files(genes_list, file_suffixes)
+
+    # filter files based on contents, must be FASTA with sequences
+    genes_list = filter_non_fasta(genes_list)
+
+    # count number of sequences and mean length per gene
+    genes_info = gene_seqs_info(genes_list)
```

Comment:
A minor suggestion of adding multiprocessing here, as well as some feedback for the terminal.